cmd/tiller/tiller.go

```diff
@@ -215,26 +225,30 @@ func start() {
 
 	srvErrCh := make(chan error)
 	probeErrCh := make(chan error)
-	go func() {
+	done := make(chan struct{})
```

Comment:
Is this tiller upstream or a change needed for rancher? also what does it do?

Reply:
It's ours. Stops a race condition with registering before we register...

Reply:
if youre going to gate the second go routine from running until the first is done why not just combine into one or call the second one from the first?